src/cloud-api-adaptor/pkg/userdata/provision.go

```diff
@@ -33,7 +33,7 @@ type Config struct {
 }
 
 func NewConfig(aaConfigPath, agentConfig, authJsonPath, daemonConfigPath, cdhConfig string, fetchTimeout int) *Config {
-	cfgPaths := paths{aaConfigPath, agentConfig, authJsonPath, cdhConfig, daemonConfigPath}
+	cfgPaths := paths{aaConfigPath, authJsonPath, cdhConfig, daemonConfigPath, agentConfig}
```

Comment:
@bpradipt so the order of files in `paths` matter?

Reply:
Should we consider adding field names to `paths` for clarity to try and avoid this in future?

Reply:
This ^ or leave an in-code comment explaining the order matters.

Reply:
```suggestion
	cfgPaths := paths{aaConfig: aaConfigPath, agentConfig: agentConfig, authJson: authJsonPath, cdhConfig: cdhConfig, daemonConfig: daemonConfigPath}
```

Reply:
This is unambiguous